src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -59,6 +60,9 @@ public class WebJobsScriptHostService : IHostedService, IScriptHostManager, IDis
                 _healthCheckWindow = new SlidingWindow<bool>(_healthMonitorOptions.Value.HealthCheckWindow);
                 _hostHealthCheckTimer = new Timer(OnHostHealthCheckTimer, null, TimeSpan.Zero, _healthMonitorOptions.Value.HealthCheckInterval);
             }
+
+            // shutdown is scoped to a single host instance
+            _shutdownSentinelFilePath = Path.Combine(_applicationHostOptions.CurrentValue.LogPath, "Host", $"{Utility.GetInstanceId()}-{ScriptConstants.ShutdownSentinelFileName}");
```

Comment:
Chose an instance based file approach here as the signal, rather than trying to play games with the AppDomainId/count. AppDomain count may go up for other reasons than restarts, e.g. language worker recycles. The instance prefix on this marker file means only the affected instance will recognize it.